plugins/jsconnect/class.jsconnect.plugin.php

```diff
@@ -676,6 +689,9 @@ protected function settings_addEdit($sender, $args) {
         $sender->addJsFile('jsconnect-settings.js', 'plugins/jsconnect');
 
         $client_id = $sender->Request->get('client_id');
+
+        Gdn::cache()->remove(self::getProviderSqlCacheKey($client_id));
```

Comment:
This is maybe a little premature. I think you're going to be purging the cache whenever the settings page is opened, instead of when the provider is actually updated. What do you think about moving this down a few lines, after a successful save?

```php
if ($form->save(['ID' => $client_id])) {
    Gdn::cache()->remove(self::getProviderSqlCacheKey($client_id));
    $sender->setRedirectTo('/settings/jsconnect');
}
```

Reply:
Ok. Done.